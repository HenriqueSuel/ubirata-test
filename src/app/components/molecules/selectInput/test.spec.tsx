import { render, screen, fireEvent } from '@testing-library/react';
import { vi } from 'vitest';

import { SelectInput } from './';

describe('SelectInput', () => {
  const options = [
    { title: 'Option 1', value: '1' },
    { title: 'Option 2', value: '2' },
    { title: 'Option 3', value: '3' },
  ];

  it('should render label and options correctly', () => {
    const label = 'Select an option';
    render(
      <SelectInput
        label={label}
        options={options}
        onSelected={() => { }}
      />
    );

    const labelElement = screen.getByText(label);
    expect(labelElement).toBeInTheDocument();

    options.forEach((option) => {
      const optionElement = screen.getByText(option.title);
      expect(optionElement).toBeInTheDocument();
    });
  });

  it('should call the selected callback when an option is selectedd', () => {
    const label = 'Select an option';
    const onSelectedMock = vi.fn();
    render(
      <SelectInput
        label={label}
        options={options}
        onSelected={onSelectedMock}
      />
    );

    const selectElement = screen.getByLabelText(label);

    fireEvent.change(selectElement, { target: { value: options[1].value } });

    expect(onSelectedMock).toHaveBeenCalledTimes(1);
    expect(onSelectedMock).toHaveBeenCalledWith(options[1].value);
  });
});
