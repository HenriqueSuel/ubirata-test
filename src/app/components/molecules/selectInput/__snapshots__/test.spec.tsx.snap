// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SelectInput > should render SelectInput 1`] = `
<div
  class="mx-4"
>
  <label
    for="Select an option"
  >
    Select an option
  </label>
  <select
    class="block appearance-none bg-gray-200 border border-gray-200 text-gray-700 py-1 px-2 pr-7 rounded leading-tight focus:outline-none focus:bg-white focus:border-gray-500"
    id="Select an option"
  >
    <option
      value="1"
    >
      Option 1
    </option>
    <option
      value="2"
    >
      Option 2
    </option>
    <option
      value="3"
    >
      Option 3
    </option>
  </select>
</div>
`;
